src/bsp/src/stm32/src/PersistedStorageManager.cpp

```diff
@@ -0,0 +1,46 @@
+#include "PersistedStorageManager.h"
+#include <DefaultSettings.h>
+#include <Task.h>
+#include <cstdio>
+#include <cstring>
+#include <hal/hal_flash.h>
+
+#define DEFAULT_UUID 1
+
+PersistedStorageManager::PersistedStorageManager(ILogger& logger) : m_logger(logger) {}
+
+void PersistedStorageManager::loadFromFlash() {
+    std::memcpy(&m_storage, reinterpret_cast<const void*>(USER_DATA_FLASH_START_ADDRESS),
+                sizeof(m_storage));
+
+    if (UUID_OVERRIDE != UINT16_MAX) {
```

Comment:
I think we could use 0 instead of UINT16_MAX for this case since it's already reserved. The user shouldn't set it's id to 0 anyway

Reply:
It's because erased flash has a value of 0xFF not 0x0, so this catches a board which has not had an ID set yet

Reply:
It makes sense for the read value of `m_storage.m_uuid` but for the UUID_OVERRIDE, since it's a build constant, checking it against 0 would make more sense than UINT16_MAX

Reply:
Changed for 0